pkg/kubelet/apis/kubeletconfig/v1beta1/defaults.go

```diff
@@ -63,118 +82,95 @@ func SetDefaults_KubeletConfiguration(obj *KubeletConfiguration) {
 	}
-	if obj.EnableServer == nil {
-		obj.EnableServer = boolVar(true)
-	}
-	if obj.FileCheckFrequency == zeroDuration {
-		obj.FileCheckFrequency = metav1.Duration{Duration: 20 * time.Second}
+	if obj.HealthzPort == nil {
+		obj.HealthzPort = utilpointer.Int32Ptr(10248)
 	}
 	if obj.HealthzBindAddress == "" {
 		obj.HealthzBindAddress = "127.0.0.1"
 	}
-	if obj.HealthzPort == nil {
-		obj.HealthzPort = utilpointer.Int32Ptr(10248)
+	if obj.OOMScoreAdj == nil {
+		obj.OOMScoreAdj = utilpointer.Int32Ptr(int32(qos.KubeletOOMScoreAdj))
 	}
-	if obj.HTTPCheckFrequency == zeroDuration {
-		obj.HTTPCheckFrequency = metav1.Duration{Duration: 20 * time.Second}
+	if obj.StreamingConnectionIdleTimeout == zeroDuration {
+		obj.StreamingConnectionIdleTimeout = metav1.Duration{Duration: 4 * time.Hour}
+	}
+	if obj.NodeStatusUpdateFrequency == zeroDuration {
+		obj.NodeStatusUpdateFrequency = metav1.Duration{Duration: 10 * time.Second}
 	}
 	if obj.ImageMinimumGCAge == zeroDuration {
 		obj.ImageMinimumGCAge = metav1.Duration{Duration: 2 * time.Minute}
 	}
 	if obj.ImageGCHighThresholdPercent == nil {
 		// default is below docker's default dm.min_free_space of 90%
-		temp := int32(85)
-		obj.ImageGCHighThresholdPercent = &temp
+		obj.ImageGCHighThresholdPercent = utilpointer.Int32Ptr(85)
 	}
 	if obj.ImageGCLowThresholdPercent == nil {
-		temp := int32(80)
-		obj.ImageGCLowThresholdPercent = &temp
-	}
-	if obj.MaxOpenFiles == 0 {
-		obj.MaxOpenFiles = 1000000
+		obj.ImageGCLowThresholdPercent = utilpointer.Int32Ptr(80)
 	}
-	if obj.MaxPods == 0 {
-		obj.MaxPods = 110
+	if obj.VolumeStatsAggPeriod == zeroDuration {
+		obj.VolumeStatsAggPeriod = metav1.Duration{Duration: time.Minute}
 	}
-	if obj.PodPidsLimit == nil {
-		temp := int64(-1)
-		obj.PodPidsLimit = &temp
+	if obj.CgroupsPerQOS == nil {
+		obj.CgroupsPerQOS = utilpointer.BoolPtr(true)
 	}
-	if obj.NodeStatusUpdateFrequency == zeroDuration {
-		obj.NodeStatusUpdateFrequency = metav1.Duration{Duration: 10 * time.Second}
+	if obj.CgroupDriver == "" {
+		obj.CgroupDriver = "cgroupfs"
 	}
 	if obj.CPUManagerPolicy == "" {
 		obj.CPUManagerPolicy = "none"
 	}
 	if obj.CPUManagerReconcilePeriod == zeroDuration {
-		obj.CPUManagerReconcilePeriod = obj.NodeStatusUpdateFrequency
+		// Keep the same as default NodeStatusUpdateFrequency
+		obj.CPUManagerReconcilePeriod = metav1.Duration{Duration: 10 * time.Second}
 	}
-	if obj.OOMScoreAdj == nil {
-		temp := int32(qos.KubeletOOMScoreAdj)
-		obj.OOMScoreAdj = &temp
-	}
-	if obj.Port == 0 {
-		obj.Port = ports.KubeletPort
+	if obj.RuntimeRequestTimeout == zeroDuration {
+		obj.RuntimeRequestTimeout = metav1.Duration{Duration: 2 * time.Minute}
 	}
-	if obj.ReadOnlyPort == nil {
-		obj.ReadOnlyPort = utilpointer.Int32Ptr(ports.KubeletReadOnlyPort)
+	if obj.HairpinMode == "" {
+		obj.HairpinMode = PromiscuousBridge
 	}
-	if obj.RegistryBurst == 0 {
-		obj.RegistryBurst = 10
+	if obj.MaxPods == 0 {
+		obj.MaxPods = 110
```

Comment:
What are the guidelines around modifying defaults like MaxPods? It seems like it shouldn't be a problem to increase it, but does that change once it's part of the API?

Reply:
Yes. Values applied to omitted keys must remain stable; when you write a v1beta1 config, the meaning of that config must not change across Kubelet versions. Changing any default value requires an API version increase.